Add "nogil" Python v3.9.10

colesbury opened the pull request:
It would be great if users could install the "nogil" fork of CPython through pyenv. This adds the build description for the latest version of the "nogil" fork.

See also https://github.com/colesbury/nogil/issues/33 and thanks to @Riatre and @thedrow

native-api commented:
AFAICS, [this fork is blessed by the core team](https://lukasz.langa.pl/5d044f91-49c1-4170-aed1-62b6763e6ad0/) and is actually planned for merging into the core sometime in the future after other related changes are made -- in particular, backward compatibility is ensured. Apparently, it has solved the catastrophic single-threaded code slowdown problem that had killed the previous efforts. So this isn't some one-shot unusable-in-real-life pet build, it's very likely to actually lead somewhere.

As such, I have no objections.

@pyenv/pyenv-core-maintainers , any other input?

native-api reviewed (approved):
[Claims to have averted the wheel incompatibility problem by including a custom modified Pip](https://github.com/colesbury/nogil#packages). So it seems to not require any special treatment, we're good.

yyuu reviewed (approved):
cool 👍

native-api commented:
@colesbury , thank you for your effort and the PR! Hopefully, availability via Pyenv will help attract more input, real-life use and contributions for your initiative!

colesbury commented:
@native-api Thanks! Pyenv will make it a lot easier for people to try out the fork.

To be clear, the fork isn't blessed by the core team. It's under consideration, but there are no plans to merge it into CPython yet. Getting more people to use it (and pyenv helps with that) will help with addressing concerns about compatibility and, I think, also show the benefit in real-life applications.
